lib/ActivityStreams.js

```diff
@@ -67,6 +68,12 @@ function ActivityStreams(options = {}) {
     preview: false,
   };
 
+  // This lets us know whether or not this is the first time someone has
+  // installed the add-on, so we can wait for previews to come back
+  simpleStorage.storage.waitForFirstPreviews = {};
```

Comment:
discussed with @k88hudson in person. It might be preferable to show the load indicator if the query cache is empy rather than to rely on prefs.

It is less error prone. The only edge case to watch out for is when a history is truly empty.

I.e. the loading indicators should show if history is not empty and the query cache is empty.